Lib/test/test_bytes.py

```diff
@@ -459,6 +459,12 @@ def test_fromhex(self):
         self.assertRaises(ValueError, self.type2test.fromhex, '\x00')
         self.assertRaises(ValueError, self.type2test.fromhex, '12   \x00   34')
 
+        # For odd number of character(s)
+        for value in ("a", "a ", " a"," a ", "aaa", "aaa ", " aaa", " aaa ",  " aa a "):
```

Comment:
What should be the behaviour of `fromhex(" aa a ")`? What message should it throw?

1.fromhex() arg must be of even length
2.non-hexadecimal number found in fromhex() arg at position

Reply:
Please review again and let me know @picnixz

Reply:
The error "must be of even length" would be confusing since it *is* of even length, so I'd say "non-hexadecimal number found in fromhex() arg at position" so that users could remove the spaces *and then* they would see that there is an odd number of *valid* characters. 

Since we are leaning towards improving user experience, having a confusing error message isn't really nice (namely, let's not replace a confusing message by another confusing one).

Reply:
I agree with your logic.

Reply:
I think we should change the error message to say "must contain an even number of hexadecimal digits" which will help with the first thing picnixz mentions, since that wording indicates we're no longer talking about string length

I don't have a strong opinion on what error message " aa a " raises. [It looks like an intentional feature that it disallow pairs with space between them](https://github.com/python/cpython/commit/0b9b9e04830c40c7866b4ba23d6fa1a0aa80f039#diff-58df2535f64e3dcfb0677a7f543924d976fd982b595f8e34902a73f3deae40ebR410), so I'm inclined to keep this example raising "non-hexadecimal number found in fromhex()"

Reply:
Which one you prefer? Or something else?

1. `s/"fromhex() arg must be of even length"/"must contain an even number of hexadecimal digits"`
2. `s/"fromhex() arg must be of even length"/"fromhex() arg must contain an even number of hexadecimal digits"`

Reply:
second one was what I was thinking